### PR TITLE
Add mode to cosmos signing input

### LINKED ISSUE #819: Add block type in TW_Cosmos_Proto_SigningInput
**Describe the bug**
Currently `TW_Cosmos_Proto_SigningInput` does not consistent `block` 

**Additional context**
```
{
   "mode":"block",
   "tx":{
      "fee":{
         "amount":[
            {
               "amount":"1000",
               "denom":"uatom"
            }
         ],
         "gas":"200000"
      },
      "memo":"p2p <> ztake",
      "msg":[
         {
            "type":"cosmos-sdk/MsgSend",
            "value":{
               "amount":[
                  {
                     "amount":"10000",
                     "denom":"uatom"
                  }
               ],
               "from_address":"cosmos135qla4294zxarqhhgxsx0sw56yssa3z0f78pm0",
               "to_address":"cosmos135qla4294zxarqhhgxsx0sw56yssa3z0f78pm0"
            }
         }
      ],
      "signatures":[
         {
            "pub_key":{
               "type":"tendermint/PubKeySecp256k1",
               "value":"ApGCXwby9foj0IAqOYvmjI+Sd2qdGVdyI1h+CiIFY8xF"
            },
            "signature":"lDR8o2J+uyTicdcXMX20PN0kH6i8Xzb9ieEeVCzggGBPdB3Jp4tsxeMplpW0raOpeFfRSab6a36hxQhOJFbiOQ=="
         }
      ],
      "type":"cosmos-sdk/MsgSend"
   }
}
```

This is a current output, but there is no to way to set `block`

### PR DESCRIPTION
**Pull request by vikmeup**
Fixes #819

### CODE REVIEW

File: src/proto/Cosmos.proto [hunk 1]
```diff
@@ -13,6 +13,11 @@ message Fee {
     uint64 gas = 2;
 }
 
+enum BroadcastMode {
+    BLOCK = 0;
+    ASYNC = 1;
```
Comment (vikmeup) [inline src/proto/Cosmos.proto]: add `Sync` as `sync` and make it default

File: tests/Cosmos/SignerTests.cpp [hunk 2]
```diff
@@ -66,3 +66,44 @@ TEST(CosmosSigner, SignTx) {
...
 }
+
+TEST(CosmosSigner, SignTxWithMemo) {
+    auto input = Proto::SigningInput();
+    input.set_account_number(1037);
+    input.set_chain_id("gaia-13003");
+    input.set_memo("");
+    input.set_sequence(8);
+    input.set_mode(Proto::BroadcastMode::ASYNC);
+    // input.set_memo("p2p <> ztake");
```
Comment (vikmeup) [inline tests/Cosmos/SignerTests.cpp]: `p2p <> ztake ` won't work

File: src/Cosmos/Serialization.cpp [hunk 3]
```diff
@@ -182,5 +182,9 @@ json Cosmos::transactionJSON(const Proto::SigningInput& input, const Data& signa
             signatureJSON(signature, Data(publicKey.bytes))
         })}
     };
-    return broadcastJSON(tx);
+    string mode = "block";
```
Comment (vikmeup) [inline src/Cosmos/Serialization.cpp]: make `sync`

File: src/proto/Cosmos.proto (hunk 1 above)
Comment (hewigovens) [reply src/proto/Cosmos.proto]: have to change all old tests?, `block` is same as `sync`

File: src/Cosmos/Serialization.cpp [hunk 4]
```diff
@@ -24,10 +24,24 @@ const string TYPE_PREFIX_MSG_REDELEGATE = "cosmos-sdk/MsgBeginRedelegate";
 const string TYPE_PREFIX_MSG_WITHDRAW_REWARD = "cosmos-sdk/MsgWithdrawDelegationReward";
 const string TYPE_PREFIX_PUBLIC_KEY = "tendermint/PubKeySecp256k1";
 
-static json broadcastJSON(json& jsonObj) {
+static string broadcastMode(Proto::BroadcastMode mode) {
+    string s = "sync";
```
Comment (madcake) [inline src/Cosmos/Serialization.cpp]: bad naming

Comment (hewigovens) [reply src/Cosmos/Serialization.cpp]: Thanks, just return string

**Pull request merged**